src/drivers/github.js

```diff
@@ -713,6 +722,72 @@ class Github {
   get userName() {
     return 'GitHub Action';
   }
+
+  async checkWorkflowForTimeoutMinutes() {
+    const WARNINGS = [];
+    const _DIR = '.github/workflows/';
+    const yaml = require('js-yaml');
+    const files = await fs.readdir(_DIR);
+    for (const file of files) {
+      const rawStr = await fs.readFile(`${_DIR}${file}`, 'utf8');
+      const doc = yaml.load(rawStr);
+      Object.keys(doc.jobs).forEach((job) => {
+        // does job contain timeout?
+        const timeoutVal = doc.jobs[job]['timeout-minutes'];
+        if (timeoutVal === undefined) return;
```

Comment:
well undefined `timeout` is also a problem right?

- timeout 4320: raise
- timeout undefined: raise
- timeout 50400: ok

Reply:
I think there are plenty of use case where cml runner instances last less that 6hrs and `timeout-minutes` is never required. Hence my effort to not bother those users with warning messages :)

Reply:
✔️ `timeout 50400: noop, all ok`

Reply:
✔️ `timeout 4320: raise`
- I disagree on `timeout undefined: raise`

✔️ / ➖ `timeout != 50400: maybe warn?` last timeout condition checked is 4320 / so if:
  1. timeout is set
  2. timeout is not 50400
  3. timeout is not 4320
  4. check for cml usage, then raise